### PR TITLE
[ISSUE #3949] support Tencent Polaris Registration Center

### PR DESCRIPTION
**Pull request by damonxue**
As for support pushing configuration to Polaris:
 [#3949 ](https://github.com/apache/shenyu/issues/3949).
 And pr.#4410 [#4410 ](https://github.com/apache/shenyu/pull/4410)
 

 - [x] Add shenyu-register-client-server/shenyu-register-client-server-polaris module .
 - [x] Add shenyu-register-client/shenyu-register-client-polaris module .
 - [x] Add shenyu-register-instance/shenyu-register-instance-polaris module .
 - [x] Add shenyu-sync-data-center/shenyu-sync-data-polaris module .
 - [x] Add shenyu-spring-boot-starter/shenyu-spring-boot-starter-sync-data-center/shenyu-spring-boot-starter-sync-data-polaris module .

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by damonxue** [comment]
@dragon-zhang @mahaitao617 pls review.

File: shenyu-registry/shenyu-registry-polaris/src/main/java/org/apache/shenyu/registry/polaris/PolarisInstanceRegisterRepository.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * Persist instance.
+     * @param instance instance
+     */
+    @Override
+    public void persistInstance(final InstanceEntity instance) {
+        final InstanceRegisterRequest req = new InstanceRegisterRequest();
+        req.setInstanceId(buildInstanceNodeName(instance));
+        req.setWeight(1);
```
Comment (moremind) [inline shenyu-registry/shenyu-registry-polaris/src/main/java/org/apache/shenyu/registry/polaris/PolarisInstanceRegisterRepository.java:86]: why 1?

**Review by moremind — CHANGES_REQUESTED** [review]

File: shenyu-registry/shenyu-registry-polaris/src/main/java/org/apache/shenyu/registry/polaris/PolarisInstanceRegisterRepository.java (hunk 1 above)
Comment (damonxue) [reply shenyu-registry/shenyu-registry-polaris/src/main/java/org/apache/shenyu/registry/polaris/PolarisInstanceRegisterRepository.java:86]: refer to 'org.apache.shenyu.registry.nacos.NacosInstanceRegisterRepository#persistInstance';
`
    org.apache.shenyu.registry.nacos.NacosInstanceRegisterRepository#persistInstance

    @Override
    public void persistInstance(final InstanceEntity instance) {
        try {
            Instance inst = new Instance();
            inst.setWeight(1.0d);
            inst.setEphemeral(true);
            inst.setIp(instance.getHost());
            inst.setPort(instance.getPort());
            inst.setInstanceId(buildInstanceNodeName(instance));
            inst.setServiceName(instance.getAppName());
            namingService.registerInstance(instance.getAppName(), groupName, inst);
            LOGGER.info("nacos client register success: {}", inst);
        } catch (NacosException e) {
            throw new ShenyuException(e);
        }
    }
`

**Comment by moremind** [comment]
pls add license file

**Review by moremind — APPROVED** [review]

**Pull request merged**